docs/dxp/7.x/en/liferay-internals/extending-liferay/overriding-osgi-services/reconfiguring-components-to-use-your-osgi-service/resources/liferay-m1t1.zip/m1t1-impl/src/main/java/com/acme/m1t1/internal/M1T1Impl.java

```diff
@@ -0,0 +1,16 @@
+package com.acme.m1t1.internal;
+
+import com.acme.m1t1.M1T1;
+
+import org.osgi.service.component.annotations.Component;
+
+@Component(immediate = true, service = M1T1.class)
```

Comment:
I removed `immediate = true` because of the [CheatSheet](https://gist.github.com/jhinkey/b7bf0be9946246502c9a32ef35626afc#component-annotations):

"Don't use the `immediate = true property`. Use it only if testing shows that you need it. You should only need it if the class must perform an initialization before it's used."